agent/agent.go

```diff
@@ -1485,15 +1485,30 @@ func writeFileAtomic(path string, contents []byte) error {
 		return err
 	}
 	if _, err := fh.Write(contents); err != nil {
+		logger.Printf("[INFO] Writing to temp file at %v failed, deleting...\n", tempPath)
```

Comment:
These log prints don't need a trailing newlines, but I'd get rid of these altogether since we already bubble the error up to the caller.